### ISSUE TITLE
conflict between scene_box aabb with disable_scene_contraction in instant-ngp / OccGridEstimator

### ISSUE THREAD

**Issue by hoanhle** [nerfstudio-project/nerfstudio#3405]
`OccGridEstimator` uses different levels of `aabbs` to know where to sample the points from with difference levels of details

```
        # multiple levels of aabbs
        aabbs = torch.stack(
            [_enlarge_aabb(roi_aabb, 2**i) for i in range(levels)], dim=0
        )
...
  intervals, samples = traverse_grids(
            rays_o,
            rays_d,
            self.binaries,
            self.aabbs,
            near_planes=near_planes,
            far_planes=far_planes,
            step_size=render_step_size,
            cone_angle=cone_angle,
        )
        t_starts = intervals.vals[intervals.is_left]
        t_ends = intervals.vals[intervals.is_right]
```

But those samples outside the original `aabb` is not used anyway if we disable the scene contraction
```
        if self.spatial_distortion is not None:
            positions = ray_samples.frustums.get_positions()
            positions = self.spatial_distortion(positions)
            positions = (positions + 2.0) / 4.0
        else:
            positions = SceneBox.get_normalized_positions(ray_samples.frustums.get_positions(), self.aabb)
```

What am I missing here? Shouldn't we instead do something like `_enlarge_aabb` with factor `0.5` when the scene contraction is disabled?

**Comment by liruilong940607** [comment]
when using `OccGridEstimator` with `levels=1`, the `roi_aabb` is exactly just the `self.aabb` used to normalize the `positions`. If `levels>1`, then the `self.aabb` should equal to `_enlarge_aabb(roi_aabb, 2**(levels-1))`

**Comment by hoanhle** [comment]
@liruilong940607, `self.aabb `is used to select the points that belong to the `roi`, correct? In the toy example below (with `levels = 2`), it seems that most of the points in the space in the outer occupancy grid don't get selected later in `NerfactoField`. I feel like I might be missing something simple here. Could you help clarify the intuition behind this?

```
 # Make sure the tcnn gets inputs between 0 and 1.
        selector = ((positions > 0.0) & (positions < 1.0)).all(dim=-1)
        positions = positions * selector[..., None]
```

![image](https://github.com/user-attachments/assets/a8d006fe-2baf-416f-a239-7d0d1d4a8645)

**Comment by hoanhle** [comment]
@liruilong940607 tnx, i can open an PR to fix this once i have time!
